### ISSUE TITLE
Training EQTransformer model: Loss function

### ISSUE THREAD

**Issue by sfpoveda** [seisbench/seisbench#112]
Hi,

I have a question about the loss function we use to train EQTransformer. From before you recommended me to use a Binary Cross Entropy loss function instead of the classic Cross Entropy Loss function, because we are not predicting a noise output. However, I think that the Detection class is replacing the Noise class, so that we would have 3 classes still, do not? And if we have three classes, why do we use the Binary Cross Entropy loss function (for only two classes) and not the classic Cross Entropy loss function (three classes)?

**Comment by yetinam** [comment]
Hi @sfpoveda ,
when using cross-entropy loss, a required prerequisite is that the sum of all classes at each time point is 1. This is true for picks and noise, but not for the picks and the detection trace. In other words, you are predicting one distribution over picks and noise.
In contrast, for EQTransformer you need to apply binary cross-entropy to each prediction (p, s, detection separately), i.e., you look at three separate traces. For each of them, you score the prediction (e.g. P) against it's compliment (e.g. not P). The different predictions are independent of each other. In other words, binary cross entropy adds a virtual "noise" trace separately for each prediction channel. I personally found it quite instructive to write down the formulas for the two loss functions. It shows quite nicely, why cross entropy converges against a trivial minimum if the classes do not sum to 1.

**Comment by sfpoveda** [comment]
Lot of thanks, it is very clear now!
